### PR TITLE
[11.0][MIG] stock_picking_operation_quick_change

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Standard migration.

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@mreficent Can you check tests? they are failing.

**Comment by MiquelRForgeFlow** [comment]
tests :green_apple:

File: stock_picking_operation_quick_change/README.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+To use this module, you need to:
+
+#. Go to products and create one of type "Stockable".
+#. Update quantities on hand to have stock of it.
+#. Go to inventory dashboard and click on "Delivery Orders" card to do a new
+   transfer.
+#. Create a picking and select the product to do the transfer.
+#. Reserve the picking.
+#. Click on button *Change Location*, select in wizard the old location and
+   new location.
```
Comment (LoisRForgeFlow) [inline stock_picking_operation_quick_change/README.rst]: Please reproduce this steps yourself to test the module

Comment (MiquelRForgeFlow) [reply stock_picking_operation_quick_change/README.rst]: Updated. Now this button appears :)

Comment (LoisRForgeFlow) [reply stock_picking_operation_quick_change/README.rst]: :wink:

**Review by LoisRForgeFlow — APPROVED** [review]
Tested :+1:

File: stock_picking_operation_quick_change/README.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to products and create one of type "Stockable".
+#. Update quantities on hand to have stock of it.
+#. Go to inventory dashboard and click on "Delivery Orders" card to do a new
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/README.rst]: do -> create

File: stock_picking_operation_quick_change/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+To use this module, you need to:
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/readme/USAGE.rst:1]: see above comments

File: stock_picking_operation_quick_change/views/stock_picking_view.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/views/stock_picking_view.xml:2]: 2017-2018 (everywhere)

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        string='New destination location',
+        required=True,
+        domain=lambda self: self._get_allowed_location_domain(),
+    )
+    change_all = fields.Boolean(
+        string='Change All',
+        help='Check if you want change all operations without filter '
+             'by old location')
+
+    def check_forbbiden_pickings(self, pickings):
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard.py]: forbidden

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        forbidden_pickings = pickings.filtered(
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard.py]: It is kind of confusing: forbidden or allowed?
(nothing wrong in the logic though :smile:)

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            lambda x: x.state not in self._get_allowed_picking_states())
+        if forbidden_pickings:
+            raise UserError(_(
+                'You can not change operations destination location if '
+                'picking state is not in %s') % ','.join(
+                self._get_allowed_picking_states()))
+        pikings_with_chained_moves = pickings.filtered(
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard.py]: pickings

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            lambda x: x.move_lines.mapped('move_dest_ids'))
+        if pikings_with_chained_moves:
+            raise UserError(_(
+                'You cannot change destination location if any move has a '
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard.py]: 'You cannot change the destination location if any of the move already has a '

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard_view.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard_view.xml]: copyright

File: stock_picking_operation_quick_change/wizards/stock_picking_wizard_view.xml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    <!-- wizard view -->
+    <record id="stock_picking_operation_change_location_view" model="ir.ui.view">
+        <field name="name">Change Operation Destination Location</field>
+        <field name="model">stock.picking.operation.wizard</field>
+        <field name="arch" type="xml">
+            <form string="Stock Picking">
+                <div>
+                    You can change destination location in operations.
```
Comment (elicoidal) [inline stock_picking_operation_quick_change/wizards/stock_picking_wizard_view.xml]: This looks a bit awkward

**Review by elicoidal — CHANGES_REQUESTED** [review]
Thanks for the contribution

File: stock_picking_operation_quick_change/views/stock_picking_view.xml (hunk 4 above)
Comment (MiquelRForgeFlow) [reply stock_picking_operation_quick_change/views/stock_picking_view.xml:2]: why? Sergio didn't do anything this 2018 in this module or previous versions 🤔

**Comment by MiquelRForgeFlow** [comment]
@elicoidal please, update your review

**Review by novawish — APPROVED** [review]
LGTM

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
Merging as all comments have been attended

**Pull request merged**